grate/Migration/AnsiSqlDatabase.cs

```diff
@@ -354,13 +362,24 @@ private class ScriptsRunCacheItem
 
     private async Task<IDictionary<string, string>> GetAllScriptsRun()
     {
-        var sql = $@"
+
+        try
+        {
+
+            var sql = $@"
 SELECT script_name, text_hash
 FROM {ScriptsRunTable} sr
 WHERE id = (SELECT MAX(id) FROM {ScriptsRunTable} sr2 WHERE sr2.script_name = sr.script_name)
 ";
-        var results = await Connection.QueryAsync<ScriptsRunCacheItem>(sql);
-        return results.ToDictionary(item => item.script_name, item => item.text_hash);
+            var results = await Connection.QueryAsync<ScriptsRunCacheItem>(sql);
+            return results.ToDictionary(item => item.script_name, item => item.text_hash);
+
+        }
+        catch (Exception ex) when (Config?.DryRun ?? throw new InvalidOperationException("No configuration available."))
```

Comment:
I think we should maybe isolate this exception handling that is done both above ( my previous comment), and below in `GetScriptsRunCache` , into a more general concept (a proxy/wrapper of some sorts), to avoid performing almost the same exception handing many places in a situation that is in essence an expected situation (although very rare) :)

Reply:
I think I'd rather see  a better thought out impl of `dryrun` than I wrote originally rather than an attempt to arbitrarily  centralise error handling which is always going to be situation specific.

Reply:
I agree. We need to re-think it a bit. We need to look a bit more into it. I don't think a _single_ "DryRunConnection" would solve it, as there does need to be database specific ones, but we might wrap the DbConnection implementations or similar, to make the "if dryrun" logic _one_ place. 

Let's ponder a bit, and get back to it